Source/DafnyCore/AST/Grammar/ProgramParser.cs

```diff
@@ -98,12 +97,13 @@ public class ProgramParser {
   }
 
   private DfyParseResult ParseFileWithErrorHandling(DafnyOptions options,
+    bool parseAsDooFile,
```

Comment:
There's also the alternative of ensuring doo files contain only the original program text, before resolution/refinement/etc. I think this is moving in the right direction though, because eventually it would be great to not re-resolve doo files after parsing.